arklib/src/lib.rs

```diff
@@ -39,9 +38,10 @@ pub mod android {
         let file_path: &Path = Path::new(&file_name);
         trace!("Received filename: {}", file_path.display());
 
-        arklib::id::ResourceId::compute(file_size.try_into().unwrap(), file_path)
-            .crc32
-            .into()
+        let resourceId = arklib::id::ResourceId::compute(file_size.try_into().unwrap(), file_path);
+        env.new_string(format!("{:x}-{}", resourceId.crc32, resourceId.file_size))  
```

Comment:
In [persisted index PR](https://github.com/ARK-Builders/arklib/pull/29) I've implemented `Display` trait for `ResourceId`, so you can just use that.